File: HW/HW1/validPalindrome.java
package HW.HW1;

public class validPalindrome {

    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        s = s.replaceAll("[^a-zA-Z0-9]", "");
        int i = 0; int j = s.length() - 1;        
        while (i < j)
        {
            if (s.charAt(i) != s.charAt(j))
            {
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
}

File: HW/HW1/validPalindrome.py
import re

def isPalindrome(self, s: str) -> bool:
    string = re.sub("[^0-9a-zA-Z]", "", s).lower()
    i = 0
    j = len(string) - 1
    while i < j:
        if string[i] != string[j]:
            return False;
        i+=1
        j-=1
    return True;